File: library/src/main/java/com/qwert2603/permesso/internal/PermessoContentProvider.java
package com.qwert2603.permesso.internal;

import android.app.Activity;
import android.app.Application;
import android.content.ContentProvider;
import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

public final class PermessoContentProvider extends ContentProvider {
    @Override
    public boolean onCreate() {
        final Application application = (Application) getContext().getApplicationContext();
        application.registerActivityLifecycleCallbacks(new ActivityLifecycleCallbacksAdapter() {
            @Override
            public void onActivityResumed(@NonNull final Activity activity) {
                if (activity instanceof AppCompatActivity) {
                    ActivityProvider.INSTANCE.onActivityResumed((AppCompatActivity) activity);
                } else {
                    logNotAppCompatActivity(activity);
                }
            }

            @Override
            public void onActivityPaused(@NonNull final Activity activity) {
                if (activity instanceof AppCompatActivity) {
                    ActivityProvider.INSTANCE.onActivityPaused((AppCompatActivity) activity);
                } else {
                    logNotAppCompatActivity(activity);
                }
            }

            private void logNotAppCompatActivity(Activity activity) {
                Log.e("Permesso", "Activity must extend AppCompatActivity: " + activity);
            }
        });
        return true;
    }

    @Override
    public Cursor query(@NonNull Uri uri, String[] strings, String s, String[] strings1, String s1) {
        return null;
    }

    @Override
    public String getType(@NonNull Uri uri) {
        return null;
    }

    @Override
    public Uri insert(@NonNull Uri uri, ContentValues contentValues) {
        return null;
    }

    @Override
    public int delete(@NonNull Uri uri, String s, String[] strings) {
        return 0;
    }

    @Override
    public int update(@NonNull Uri uri, ContentValues contentValues, String s, String[] strings) {
        return 0;
    }
}

File: permesso_crtn/src/main/java/com/qwert2603/permesso_crtn/PermessoContentProvider.kt
package com.qwert2603.permesso_crtn

import android.app.Activity
import android.app.Application
import android.content.ContentProvider
import android.content.ContentValues
import android.database.Cursor
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity

internal class PermessoContentProvider : ContentProvider() {
    override fun onCreate(): Boolean {
        val application = context!!.applicationContext as Application

        application.registerActivityLifecycleCallbacks(object : ActivityLifecycleCallbacksAdapter() {
            override fun onActivityResumed(activity: Activity) {
                ActivityProvider.onActivityResumed(activity as AppCompatActivity)
            }

            override fun onActivityPaused(activity: Activity) {
                ActivityProvider.onActivityPaused(activity as AppCompatActivity)
            }
        })
        return true
    }

    override fun insert(p0: Uri, p1: ContentValues?): Uri? {
        return null
    }

    override fun query(p0: Uri, p1: Array<out String>?, p2: String?, p3: Array<out String>?, p4: String?): Cursor? {
        return null
    }

    override fun update(p0: Uri, p1: ContentValues?, p2: String?, p3: Array<out String>?): Int {
        return 0
    }

    override fun delete(p0: Uri, p1: String?, p2: Array<out String>?): Int {
        return 0
    }

    override fun getType(p0: Uri): String? {
        return null
    }
}